src/t_list.c

```diff
@@ -385,7 +385,53 @@ void lsetCommand(client *c) {
     }
 }
 
-void listElementsRemoved(client *c, robj *key, int where, robj *o) {
+/* A helper for replying with a list's range as a multi-bulk. For backward
+* compatibility, start needs to be less than end. The direction affects how
+* elements are iterated. */
+void addListRangeReply(client *c, robj *o, long start, long end, int direction) {
+    long rangelen, llen = listTypeLength(o);
+
+    /* Convert negative indexes. */
+    if (start < 0) start = llen+start;
+    if (end < 0) end = llen+end;
+    if (start < 0) start = 0;
+
+    /* Invariant: start >= 0, so this test will be true when end < 0.
+     * The range is empty when start > end or start >= length. */
+    if (start > end || start >= llen) {
+        addReply(c,shared.emptyarray);
+        return;
+    }
+    if (end >= llen) end = llen-1;
+    rangelen = (end-start)+1;
+
+    /* Return the result in form of a multi-bulk reply */
+    addReplyArrayLen(c,rangelen);
+    if (o->encoding == OBJ_ENCODING_QUICKLIST) {
+        listTypeIterator *iter = listTypeInitIterator(o,
+                                                      (direction == LIST_TAIL)
+                                                      ? start
+                                                      : end,
+                                                      direction);
```

Comment:
IMHO this looks bad, i'd rather have it in one line. if it's too long, we can
1. move the variable declaration to a separate line.
2. declare another variable for the start (using the ternary operator) so that in the function call we just use it.

Reply:
I picked this formatting from JavaScript - sorry! Mia culpa!

Reply:
Option 2 picked in bae0cb8